dbms/src/Flash/Mpp/MPPTask.cpp

```diff
@@ -99,7 +99,14 @@ void MPPTaskMonitorHelper::initAndAddself(MPPTaskManager * manager_, const Strin
 MPPTaskMonitorHelper::~MPPTaskMonitorHelper()
 {
     if (initialized)
+    {
+        /// Because MemoryTracker is now saved in `MPPQueryTaskSet`, it will not be destructed when
+        /// MPPTask is destructed, so need to manually reset current_memory_tracker to nullptr at the
+        /// end of the destructor of MPPTask, otherwise, current_memory_tracker may point to a invalid
+        /// memory tracker
```

Comment:
This comment makes me a little confused, because MPPQueryTaskSet will call removeMPPQueryTaskSet to destroy MPPQueryTaskSet and the MemoryTracker it holds after all MPPTasks have called unregisterTask.
However, MPPTask may still not be destructed after MPPQueryTaskSet is destructed. At this time, MPPTask will hold the shared_ptr of Query Memory Tracker

Reply:
Before this pr, memory_tracker/processlistentry is hold by mpp task, and each mpp task has its own memory_tracker/processlistentry, so when mpp task is destructed, the memory tracker will be destructed, and `current_memory_tracker` is set to nullptr in the destructor of memory tracker, after this pr, the memory tracker can only be destructed once, it mean for most of the mpp task, when mpp task is destructed, memory tracker is likely to be not destructed, so we need to manually reset `current_memory_tracker`

Reply:
It's a bit strange to reset `cur_memory_tracker` by `MPPTaskMonitorHelper` .
How about adding a new class to do this?
```
class ProcessListEntryHolder
{
std::shared_ptr<ProcessListEntry> process_list_entry;

~ProcessListEntryHolder()
{
    if likely (process_list_entry)
    {
        process_list_entry.reset();
        current_memory_tracker = nullptr;
    }
}
}
```
and put `ProcessListEntryHolder` under `MPPTaskMonitorHelper`.

Reply:
And maybe the comments can be improved..

Reply:
I aggree, update the code to add `ProcessListEntryHolder `